airbyte-cdk/python/airbyte_cdk/sources/declarative/datetime/datetime_parser.py

```diff
@@ -36,3 +52,17 @@ def format(self, dt: datetime.datetime, format: str) -> str:
             return str(int(dt.timestamp()))
         else:
             return dt.strftime(format)
+
+    def find_most_granular_timedelta(self, datetime_format: str) -> relativedelta:
+        """
+        The granularity is based on the 1989 C standard directives. If the datetime_format matches `%Y-%m-%dT%H:00:00` for example, this
```

Comment:
What is the expectation regarding a format like `%Y-%m-%dT%H:00:00`?

Reply:
[A new PR](https://github.com/airbytehq/airbyte/pull/20717) has been opened with a solution that requires user input but supports this case